crates/weaver_semconv/src/body.rs

```diff
@@ -0,0 +1,244 @@
+#[serde(rename_all = "snake_case")]
+pub enum BodySpec {
+    /// The collection of body fields associated with a body definition
+    Fields {
+        /// The collection of body fields associated with a body definition
+        #[serde(skip_serializing_if = "Vec::is_empty")]
+        fields: Vec<BodyFieldSpec>,
+    },
+    /// The body field value.
+    Value {
+        /// The body field value.
+        value: ValueSpec,
+    },
+}
+
+impl BodySpec {
+    /// Returns true if the body field is required.
+    #[must_use]
+    pub fn has_fields(&self) -> bool {
+        match self {
+            BodySpec::Fields { fields } => !fields.is_empty(),
+            BodySpec::Value { value: _ } => false,
+        }
+    }
+}
+
+/// A `BodyField` specification.
+#[derive(Serialize, Deserialize, Debug, Clone)]
+#[serde(deny_unknown_fields)]
+#[serde(rename_all = "snake_case")]
+pub struct BodyFieldSpec {
+    /// String that uniquely identifies the body field.
+    pub id: String,
+    /// Either a string literal denoting the type as a primitive or an
+    /// array type, a template type or an enum definition.
+    pub r#type: AttributeType,
+    /// A brief description of the body field.
+    pub brief: Option<String>,
+    /// Sequence of example values for the body field or single example
+    /// value. They are required only for string and string array
+    /// fields. Example values must be of the same type of the
+    /// body field. If only a single example is provided, it can directly
+    /// be reported without encapsulating it into a sequence/dictionary.
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub examples: Option<Examples>,
+    /// Specifies if the body field is mandatory. Can be "required",
+    /// "conditionally_required", "recommended" or "opt_in". When omitted,
+    /// the body field is "recommended". When set to
+    /// "conditionally_required", the string provided as <condition> MUST
+    /// specify the conditions under which the body field is required.
+    #[serde(default)]
+    pub requirement_level: RequirementLevel,
+    /// A more elaborate description of the body field.
+    /// It defaults to an empty string.
+    #[serde(default)]
+    pub note: String,
+    /// Specifies the stability of the body field.
+    /// Note that, if stability is missing but deprecated is present, it will
+    /// automatically set the stability to deprecated. If deprecated is
+    /// present and stability differs from deprecated, this will result in an
+    /// error.
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub stability: Option<Stability>,
+    /// Specifies if the body field is deprecated. The string
+    /// provided as <description> MUST specify why it's deprecated and/or what
+    /// to use instead. See also stability.
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub deprecated: Option<String>,
+}
+
+impl BodyFieldSpec {
+    /// Returns true if the body field is required.
```

Comment:
Should this also return conditionally required or not?

Reply:
This is really the `BodyFieldSpec` version of the to the `AttributeSpec` `is_required` (which is not currently checking for any conditionally required)/ So my simple answer is no. But it's also currently not directly used as we pulled out the markdown examples. So to keep them in-sync, my simple answer to this is No.